cadc-vos-server/src/main/java/org/opencadc/vospace/server/NodePersistence.java

```diff
@@ -131,6 +134,27 @@ public interface NodePersistence {
      */
     public Set<URI> getImmutableProps();
 
+    /**
+     * Get the default value of a Node Property.  This will likely be used by create operations when values are missing.
+     * @param propertyKey   The URI of the property key to get the default value for.
+     * @return  String default value for the given property key, or null if no default value is set.
+     */
+    String getDefaultPropertyValue(final URI propertyKey);
+
+    /**
+     * Check if the given caller can administer allocations.  Used by create operations.
+     * @param caller    The caller subject, used to check permissions.
+     * @return  True if the given caller can create new administer new allocations, false otherwise.
+     */
+    boolean isAdmin(Subject caller);
```

Comment:
In principle, an admin can do several things more general than create allocation:
- create a Node owned by someone else
- add or modify admin-only props (quota)
In future, it will probably be the way to modify owner if/when we need that.

So, at a minimum the method name here is more general than what we intend to allow via api keys. I see that UpdateAction also calls this, but api-key callers won';t get that far because UpdateAction doesn't switch users like CreateAction does... right now that's nominally intentional, but that mismatch will look like a bug in a few months. I can also see prepareData wanting to insure a minimum quota in future (it will know how much data it intends to transfer in) and thus wanting to modify an existing quota.

Let's discuss tomorrow in person.

Reply:
Changes submitted.